### ISSUE TITLE
cannot connect to postgresql

### ISSUE THREAD

**Issue by sandspirit2** [jhipster/generator-jhipster-nodejs#189]
**Describe the bug**

**To Reproduce**

generate and application with this jdl (https://start.jhipster.tech/jdl-studio/#/view/608d075c-c06c-4f0e-8642-4920fbe85d64)
select react and postgresql and i can't seem to be able to connect to the postgres database even after fixing the configuratin and everything 
here are 2 examples of my configurations i tried 

if (process.env.NODE_ENV === 'prod') {
  ormconfig = {
    name: 'default',
    type: 'postgres',
    database: 'technicalq',
    url: 'postgresql://postgres:admin@localhost:5432/technicalq',
    logging: false,
    synchronize: commonConf.SYNCRONIZE,
    entities: commonConf.ENTITIES,
    migrations: commonConf.MIGRATIONS,
    cli: commonConf.CLI,
    migrationsRun: commonConf.MIGRATIONS_RUN
  };
}

if (process.env.NODE_ENV === 'test') {
  ormconfig = {
    type: 'postgres',
    port: 5432,
    username: 'postgres',
    password: 'admin',
    database: 'technicalq',
    synchronize: true,
    logging: true ,
    //synchronize: commonConf.SYNCRONIZE,
    entities: commonConf.ENTITIES,
    migrations: commonConf.MIGRATIONS,
    cli: commonConf.CLI,
    migrationsRun: commonConf.MIGRATIONS_RUN
  };

if somebody can give me some pointers i would be appreciative i also installed the pg library using: npm i pg --save and created the same schema in postgres like i read iin one of the  github issues from a ear ago i think but it was about mysql 
so that's the result it just clean exists 
![image](https://user-images.githubusercontent.com/34859699/101361257-662f4580-38a7-11eb-827b-9da0515f2e3f.png)

[.yo-rc.zip](https://github.com/jhipster/generator-jhipster-nodejs/files/5653159/default.yo-rc.zip)

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

-   OS: [e.g. iOS]
-   Browser [e.g. chrome, safari]
-   Version [e.g. 22]

**NHipster configuration**

**Additional context**

**Comment by amanganiello90** [comment]
Hi @sandspirit2 thanks for the interest!
So I do not unterstand very well your issue.
> even after fixing the configuratin and everything
1) What elements do you have fixed? 

> url: 'postgresql://postgres:admin@localhost:5432/technicalq'
2) It's correct? your username is postgres and your password admin? And do you have a postgres database in your localhost:5432 with database name "techincalq"?

> if (process.env.NODE_ENV === 'test') {
3) You have to change only the url in the prod profile and not in the test profile:
```
if (process.env.NODE_ENV === 'prod') {
ormconfig = {
```

**Comment by sandspirit2** [comment]
> Hi @sandspirit2 thanks for the interest!
> So I do not understand very well your issue.
> so yeah after changing the url and everything like you mentioned the application doesn't start
> > even after fixing the configuratin and everything for some reason i can't run it in prod env
> 
> 1. What elements do you have fixed?
> 
> > url: 'postgresql://postgres:admin@localhost:5432/technicalq'
> 
> 1. Is it correct? Your username is postgres and your password admin? And do you have a postgres database in your localhost:5432 with database name "techincalq"?
> yes 
![image](https://user-images.githubusercontent.com/34859699/101367112-74349480-38ae-11eb-9d03-2a973c7303db.png)

> > if (process.env.NODE_ENV === 'test') {
> 
> 1. Eventually, you have to change only the url in the prod profile and not in the test profile:
> 
> ```
> if (process.env.NODE_ENV === 'prod') {
> ormconfig = {
> name: 'default',
> type: 'postgres',
> database: 'technicalq',
> url: 'postgresql://postgres:admin@localhost:5432/technicalq',
> ```
> 
> > if somebody can give me some pointers i would be appreciative i also installed the pg library using: npm i pg --save and
> 
> 1. Postgres (pg dependency) is already added in your **server/package.json** by the generator.

**Comment by amanganiello90** [comment]
Have you run postgres with docker compose file under your **src/main/docker/postgresql.yml** file?
So the template is: https://github.com/jhipster/generator-jhipster-nodejs/blob/main/generators/server/templates/src/main/docker/postgresql.yml.ejs

That postgres has another username/password and you have to create the database before.

**Comment by sandspirit2** [comment]
> Have you run postgres with docker compose file under your **src/main/docker/postgresql.yml** file?
> So the template is: https://github.com/jhipster/generator-jhipster-nodejs/blob/main/generators/server/templates/src/main/docker/postgresql.yml.ejs
> 
> That postgres has another username/password and you have to create the database before.

yes that's the file and i changed the username and paswords and the database accordingly

**Comment by amanganiello90** [comment]
> 
> 
> > Have you run postgres with docker compose file under your **src/main/docker/postgresql.yml** file?
> > So the template is: https://github.com/jhipster/generator-jhipster-nodejs/blob/main/generators/server/templates/src/main/docker/postgresql.yml.ejs
> > That postgres has another username/password and you have to create the database before.
> 
> yes that's the file and i changed the username and paswords and the database accordingly

And with the connection string specified:  
```
url: 'postgresql://postgres:admin@localhost:5432/technicalq',
```

Can you connect with a postgres sql client tool?

**Comment by sandspirit2** [comment]
i can connect to the dockerized postgres using pgadmin but not using the application

**Comment by amanganiello90** [comment]
I check now generating a new app with the latest version. Stay turned 💯

**Comment by sandspirit2** [comment]
> I check now generating a new app with the latest version. Stay turned 💯

ok great thank you and yes i created a schema and same issue so if you can tell me the steps you would do to connect to postgres that would be really helpful

**Comment by amanganiello90** [comment]
So I have checked and it is all run. The steps that I perfomed are the followings:

1. I generated an application (named jhipster) without client, and choosing postgresql as database. Here the yo-rc.json file:

```
{
  "generator-jhipster": {
    "jhipsterVersion": "6.8.0",
    "applicationType": "monolith",
    "baseName": "jhipster",
    "packageName": "com.jhipster.node",
    "packageFolder": "com/jhipster/node",
    "serverPort": "8081",
    "authenticationType": "jwt",
    "cacheProvider": "no",
    "enableHibernateCache": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "sqlite",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "embeddableLaunchScript": false,
    "useSass": true,
    "clientPackageManager": "npm",
    "skipClient": true,
    "creationTimestamp": 1607355548221,
    "testFrameworks": ["protractor"],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [
      {
        "name": "generator-jhipster-nodejs",
        "version": "1.3.0"
      }
    ],
    "enableTranslation": false,
    "blueprints": [
      {
        "name": "generator-jhipster-nodejs",
        "version": "1.3.0"
      }
    ]
  }
}


```

2. I changed the **src/main/docker/postgresql.yml env variables in:

```
version: '2'
services:
  jhipster-postgresql:
    image: postgres:12.1
    # volumes:
    #     - ~/volumes/jhipster/jhipster/postgresql/:/var/lib/postgresql/data/
    environment:
      - POSTGRES_USER=jhipster
      - POSTGRES_PASSWORD=jhipster
      - POSTGRES_HOST_AUTH_METHOD=trust
    ports:
      - 5433:5432

```

So my user and password is named **jhipster**. In addition I have changed the external port in 5433 (on my pc the 5432 is already used by another postgres instance)

3. I changed the url configuration in **orm.config.ts** as:

```
if (process.env.NODE_ENV === 'prod') {
    ormconfig = {
        name: 'default',
        type: 'postgres',
        database: 'jhipster',
        url: 'postgresql://jhipster:jhipster@localhost:5433/jhipster',
        logging: false,
        synchronize: commonConf.SYNCRONIZE,

```

My database name is also **jhipster** (the name of my app). To create schema and the database, only in the first run I have set the  **synchronize option to true**.

4. On windows prompt I start the application (from the base dir generated) and database with:

```
cd server
set NODE_ENV=prod&& npm start
```
And in another prompt:

```
docker-compose -f  src/main/docker/postgresql.yml up
```

I hope this explanation resolves your issue. Thanks

**Comment by sandspirit2** [comment]
ok i will check again thank you very much

**Comment by sandspirit2** [comment]
hello i finally solved the problem i was using node v14.15 after i made nvm use v12.20 everything is working fine now thank you so much for your help

**Comment by amanganiello90** [comment]
Thanks you!!! Every questions or issues are welcome!
